gemviz/select_fields_tablemodel.py

```diff
@@ -342,7 +341,20 @@ def setFields(self, fields):
             if field.selection is not None:
                 column_number = self.columnNumber(field.selection)
                 if column_number in self.checkboxColumns:
-                    self.selections[row] = field.selection
+                    self.setSelectionsItem(row, field.selection)
+
+    def selections(self):
+        """Return the plot selections dictionary."""
+        return self._selections
+
+    def setSelections(self, selections=None):
+        """Plot selections: dict(row_number=column number}"""
+        selections = selections or {}
+        self._selections = {k: v for k, v in selections.items() if v is not None}
+
+    def setSelectionsItem(self, key, value):
+        """Set the key in the plot selections dictionary."""
+        self.selections()[key] = value
```

Comment:
We should not modify the method (`self.selections`) but the attribute (`self._selections`)